### ISSUE TITLE
Server crashes when using compress middleware

### ISSUE THREAD

**Issue by sebastianwessel** [honojs/node-server#39]
Using `@hono/node-server` with `hono/compress` middleware does not work.
The server crashes

Minimal example:

```typescript
import { serve } from '@hono/node-server'
import { Hono } from 'hono'
import { compress } from 'hono/compress'

const app = new Hono()

app.use('*', compress())

app.get('/one', async (c) => {
  let body = 'one'

  for (let index = 0; index < 1000 * 1000; index++) {
    body += ' one'
  }
  return c.text(body)
})

const _server = serve({
  fetch: app.fetch,
  port: 8081,
})
```

Error output

```bash
Trace: TypeError: First parameter has member 'readable' that is not a ReadableStream.
    at assertReadableStream (/Users/example/example/core/node_modules/web-streams-polyfill/src/lib/validators/readable-stream.ts:5:11)
    at convertReadableWritablePair (/Users/example/example/core/node_modules/web-streams-polyfill/src/lib/validators/readable-writable-pair.ts:15:3)
    at ReadableStream.pipeThrough (/Users/example/example/core/node_modules/web-streams-polyfill/src/lib/readable-stream.ts:211:23)
    at /Users/example/example/core/lab/node_modules/hono/dist/cjs/middleware/compress/index.js:35:41
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/example/example/core/lab/node_modules/hono/dist/cjs/hono.js:227:50
    at async Server.<anonymous> (/Users/example/example/core/node_modules/@hono/node-server/dist/listener.js:29:20)
    at errorHandler (/Users/example/example/core/lab/node_modules/hono/dist/cjs/hono.js:43:11)
    at /Users/example/example/core/lab/node_modules/hono/dist/cjs/compose.js:80:27
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /Users/example/example/core/lab/node_modules/hono/dist/cjs/hono.js:227:50
    at async Server.<anonymous> (/Users/example/example/core/node_modules/@hono/node-server/dist/listener.js:29:20)

```

**Comment by tangye1234** [comment]
@sebastianwessel The problems has been verified.

hono/compress employs a web standard TransformStream aka CompressionStream, originating from this draft standard:
https://wicg.github.io/compression/#compression-stream

Chrome 80, Safari 16.4, Deno 1.19, Node 17, Cloudflare edge all implement either full or partial functionality for CompressionStream.

However hono node server opts-out node native stream implementations, due to this reason: https://github.com/honojs/node-server/issues/35#issuecomment-1512865889

Instead, it relys on a widely-used stream polyfill called `web-streams-polyfill` to extend support for Web Stream across all node versions. Unfortunately, this polyfill dose not implement CompressionStream nor DecompressionStream, which I assume you are using with node 17+, which picks the uncompatible native CompressionStream as a fallback.

So let's make a polyfill for our own:

Just install `compression-streams-polyfill` which is open sourced at https://github.com/101arrowz/compression-streams-polyfill

Then create a new file named `compression-polyfill.ts` with following code:
```typescript
import type { CompressionStreamConstructor, DecompressionStreamConstructor } from 'compression-streams-polyfill/ponyfill'
import { makeCompressionStream, makeDecompressionStream } from 'compression-streams-polyfill/ponyfill'

declare global {
  var CompressionStream: CompressionStreamConstructor
  var DecompressionStream: DecompressionStreamConstructor
}

global.CompressionStream = makeCompressionStream(TransformStream)
global.DecompressionStream = makeDecompressionStream(TransformStream)
```

import that file in the entrypoint, DONE

BTW, I believe the compression polyfill should not be included into the @hono/node-server module, as doing so would make this lightweight adapter excessively heavy. Compression should be applied in a more efficient manner, for instance, I personally use nginx with gzip enabled as as reverse proxy in the front.

**Comment by sebastianwessel** [comment]
Thanks, @tangye1234 and @yusukebe 

It works this way perfectly fine and should be marked as solution.

Regarding the mentioned #35 comment:
I personally would prefer smaller bundle size with lower code & polyfill overhead over raw performance.
It's up to the node.js dev team to improve it in first place. With Bun and Deno, there are now good competitors available, which can be used if raw performance is a main factor.

**Comment by tangye1234** [comment]
@sebastianwessel, I agree with you.

However, to cater to those who prioritize performance, I believe that it is currently unnecessary for this project to use the native fetch / stream of Node.js 18, the current implementation will also ensure better compatibility with Node.js 14-16.

However, for those who do not prioritize performance but desire a smaller bundle size, I am more than happy to provide a new polyfill specifically designed to remove the web-stream-polyfill and utilize the native capabilities. This polyfill will only introduce a minimal overhead and a small package size.

Please go to https://github.com/honojs/node-server/issues/35#issuecomment-1522806685

**Comment by renhiyama** [comment]
Any updates on this? I really wanna try gzip middleware on nodejs server 🥲

**Comment by yusukebe** [comment]
@renhiyama 

Don't be in a hurry. We are discussing the related issue, #35. Depending on the decision, Compress may be able to implement it.

**Comment by yusukebe** [comment]
This issue might be closed since v1.0.0.
